e2e/rook-test-framework/rook-infra-manager/rook_test_infra_manager.go

```diff
@@ -0,0 +1,323 @@
+
+	if err != nil {
+		return err
+	}
+
+	rawUpdated := bytes.Replace(raw, []byte(rookOperatorImagePodSpecTag), []byte(tag), 1)
+	rookOperator := string(rawUpdated)
+
+	_, _, exitCode := r.transportClient.CreateWithStdin(rookOperator)
+
+	if exitCode != 0 {
+		return errors.New(string("Failed to create rook-operator pod; kubectl exit code = " + string(exitCode)))
+	}
+
+	fmt.Println()
+	if !k8sHelper.IsThirdPartyResourcePresent(rookOperatorCreatedTpr) {
+		return errors.New ("Failed to start Rook Operator; k8s thirdpartyresource did not appear")
+	} else {
+		fmt.Println("Rook Operator started")
+	}
+
+	return nil
+}
+
+func createk8sRookCluster(k8sHelper *utils.K8sHelper) error {
+	raw, err := ioutil.ReadFile(podSpecPath + "/" + rookClusterFileName)
+
+	if err != nil {
+		return err
+	}
+
+	rookCluster := string(raw)
+
+	_, _, exitCode := r.transportClient.CreateWithStdin(rookCluster)
+
+	if exitCode != 0 {
+		return errors.New("Failed to create rook-cluster pod; kubectl exit code = " + string(exitCode))
+	}
+
+	if !k8sHelper.IsServiceUpInNameSpace("rook-api") {
+		fmt.Println("Rook Cluster couldn't start")
+	} else {
+		fmt.Println("Rook Cluster started")
+	}
+
+	return nil
+}
+
+func (r *rookTestInfraManager) InstallRook(tag string) (err error, client contracts.Irook_client) {
+	//Create rook operator
+	k8sHelp := utils.CreatK8sHelper()
+
+	err = createK8sRookOperator(k8sHelp, tag)
+
+	if err != nil {
+		panic(err)
+	}
+
+	time.Sleep(10 * time.Second)	///TODO: add real check here
+
+	//Create rook cluster
+	err = createk8sRookCluster(k8sHelp)
+
+	if err != nil {
+		panic(err)
+	}
+
+	time.Sleep(5 * time.Second)
+
+	//Create rook client
+	raw, err := ioutil.ReadFile(podSpecPath + "/rook-client/" + rookClientFileName)
+
+	if err != nil {
+		panic(err)
+	}
+
+	rookClient := string(raw)
+
+	_, _, exitCode := r.transportClient.CreateWithStdin(rookClient)
+
+	if exitCode != 0 {
+		return errors.New(string(exitCode)), nil
+	}
+
+	if !k8sHelp.IsPodRunningInNamespace("rook-client") {
+		fmt.Println("Rook Client couldn't start")
+	} else {
+		fmt.Println("Rook Client started")
+	}
+
+	return nil, nil
+}
+
+func (r *rookTestInfraManager) isContainerRunning(containerId string) bool {
+	dockerClient := r.dockerContext.Get_DockerClient()
+
+	_, stdErr, _ := dockerClient.ExecuteCmd([]string{"ps", "--filter", "status=running", "--filter", "id=" + containerId, "--format", "\"{{.ID}}\""})
+
+	return strings.EqualFold(stdErr, containerId)
+}
+
+func (r rookTestInfraManager) TearDownInfrastructureCreatedEnvironment() error {
+	dindScriptName := getDindScriptName(r.k8sVersion)
+
+	cmdOut := utils.ExecuteCommand(objects.Command_Args{Command:"chmod", CmdArgs:[]string{"+x",  curdir + "/" + scriptsPath + "/" + dindScriptName}})
+
+	if cmdOut.Err != nil || cmdOut.ExitCode != 0 {
+		panic(errors.New("Failed to chmod script"))
+	}
+
+	cmdOut = utils.ExecuteCommand(objects.Command_Args{Command:  curdir + "/" + scriptsPath + "/" + dindScriptName, CmdArgs:[]string{ "up"}})
```

Comment:
why is the teardown function calling `up`? shouldn't it be calling `down` or `clean` (or both)?